### DISCUSSION TITLE
Smeared cracking rotation with respect to a principal axis

### DISCUSSION THREAD

**Discussion by aashiquear** [idaholab/moose#18786] [Q&A Modules: Solid mechanics] [answered]
I like to model smeared cracking of a spherical particle by modeling a plane (half circle) in RZ coordinate. Now, in my understanding ComputeSmearedCrackingStress rotate the crack stiffness towards the principal axis (in my case r or z or theta) direction. But to model the half-circle as a spherical particle, the principal direction at each integration point needs to be rotated at an angle delta with respect to the r direction for the smeared cracking stiffness ratio. 

delta would be a variable at each quadratic point. I am wondering how can I input delta into ComputeSmearedCrackingStress to rotate with respect to the r axis? 

![direction](https://user-images.githubusercontent.com/3649654/132402883-27470588-507e-4bdd-aa43-754502f1b750.jpg)

**Comment by GiudGiud** [comment]
@ngrilli @jiangwen84

**Comment by hugary1995** [comment, marked as answer]
I don't believe you can do this without modifying the source code. Look at the method around [https://github.com/idaholab/moose/blob/7df3b095f66fb205a46d4afc6aa4a23e4b394a95/modules/tensor_mechanics/src/materials/ADComputeSmearedCrackingStress.C#L287](https://github.com/idaholab/moose/blob/7df3b095f66fb205a46d4afc6aa4a23e4b394a95/modules/tensor_mechanics/src/materials/ADComputeSmearedCrackingStress.C#L287) where we perform the rotation.

@bwspenc knows about the smeared crack models in moose better than anyone.
